src/dbnode/storage/index/block.go

```diff
@@ -561,6 +528,9 @@ func (b *block) QueryWithIter(
 		}
 	}
 
+	iter.AddSeries(size)
```

Comment:
@ryanhall07 @robskillington Just verifying with you as I read this code multiple times, and I think I found a bug:
size is the number of elements in `results` as effectively returned originally here: 
```
func (r *results) AddDocuments(batch []doc.Document) (int, int, error) {
	r.Lock()
	err := r.addDocumentsBatchWithLock(batch)
	size := r.resultsMap.Len()
	docsCount := r.totalDocsCount + len(batch)
	r.totalDocsCount = docsCount
	r.Unlock()
	return size, docsCount, err
}
```

Since `results` only keeps growing, as I haven't seen any reset to this object, it means that when you call iter.Counts() to get the seriesMatched, you will get a wrong number, since the AddSeries(size) keeping adding up total and not adding delta. Maybe the correct way is to measure results before adding and after to get that delta?

Reply:
yea this looks like a bug.